musig/src/errors.rs

```diff
@@ -0,0 +1,46 @@
+#[derive(Debug, PartialEq)]
+pub enum MusigError {
```

Comment:
I suggest to use `thiserror` and derive `thiserror::Error` here, as it more high-level way to define errors rather than providing methods for descriptions and implementing `std::error::Error` manually.